app/routes/onboarding.tsx

```diff
@@ -1,17 +1,132 @@
-import { useLoaderData } from '@remix-run/react';
-import { json } from '@remix-run/node';
-import UserOnboarding from '~/components/userOnboarding';
+import { FC, useEffect, useState } from 'react';
+import { useLoaderData, useActionData, Outlet } from '@remix-run/react';
+import { json, redirect, ActionFunction } from '@remix-run/node';
+import axios from 'axios';
+
+interface UserOnboardingInterface {
+  apiHost: string;
+  authToken: string;
+}
 
 export async function loader() {
   return json({ apiHost: process.env.API_HOST, authToken: process.env.AUTH_TOKEN });
 }
 
-const OnboardingPage = () => {
-  const { apiHost, authToken } = useLoaderData<typeof loader>();
+type LoaderData = {
+  apiHost: string;
+  authToken: string;
+};
+
+export const action: ActionFunction = async ({ request, params }) => {
+  const formData = Object.fromEntries(await request.formData());
+  //   console.log('fullname:', formData);
+  const url = `${process.env.API_HOST}/api/v1/users/self`;
+  const { username, firstname, lastname, timezone } = formData;
+  //   const firstname = (fullname as string).split(' ')[0];
+  //   const lastname = (fullname as string).split(' ')[1];
+  console.log(formData);
+
+  const errors = {
+    username: username ? null : 'Username is required',
+    firstname: firstname ? null : 'firstname is required',
+    lastname: lastname ? null : 'lastname is required',
+    timezone: timezone ? null : 'Timezone is required',
+  };
+
+  const hasErrors = Object.values(errors).some((errorMessage) => errorMessage);
+  if (hasErrors) {
+    return json(errors);
+  }
+
+  try {
+    console.log('REQUEST:', request.headers.get('Cookie'));
+    const cookieHeader = request.headers.get('Cookie');
+    await axios.patch(url, formData, {
```

Comment:
Can we create a function called patch as a common wrapper which will accept params so that we can reuse any network request?

Reply:
yes, i think we should have a new issue created for that